### PR TITLE
Add utility to dump historical data to InfluxDB.

### PR DESCRIPTION
**Pull request by qwandor**
I've made this part of the `mijia-homie` crate so that it can share the sensor name reading code.

### CODE REVIEW

File: mijia-homie/src/mijia-history-influx.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
+//! Utility program to dump historical data from sensors to InfluxDB.
+
+#[allow(dead_code)]
+mod config;
+mod mijiahistoryconfig;
```
Comment (alsuren) [inline mijia-homie/src/mijia-history-influx.rs:5]: mijia_history_config

File: mijia-homie/src/mijia-history-influx.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    for sensor in sensors.iter() {
+        if let Some(name) = names.get(&sensor.mac_address) {
+            log::info!("Connecting to {} ({})", name, sensor.mac_address);
+            if let Err(e) = session.bt_session.connect(&sensor.id).await {
+                log::error!("Failed to connect to {}: {:?}", name, e);
+                continue;
+            }
+
+            log::info!("Reading history...");
+            let history = session.get_all_history(&sensor.id).await?;
```
Comment (alsuren) [inline mijia-homie/src/mijia-history-influx.rs:46]: Assert that the clock is vaguely synchronised || handle clock drift?

File: mijia-homie/src/mijia-history-influx.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+fn point_for_record(
+    measurement: &str,
+    mac_address: &MacAddress,
+    name: &str,
+    record: &HistoryRecord,
+) -> Point {
+    Point::new(measurement)
+        .add_timestamp(
+            record
+                .time
```
Comment (alsuren) [inline mijia-homie/src/mijia-history-influx.rs:91]: Do these time readings happen to align nicely on the hour, according to the sensor's clock?

**Review by alsuren — APPROVED** [review]
Nice work.

This would allow you to backfill a day's data for as many sensors as you want, at a resolution of one reading per hour. I guess this is assuming that the sensor clock is correct to within some reasonable margin (about 10 minutes?). @IkeLutra will be pleased.

If you run it as a cronjob every hour, will influx delicate the readings correctly? I have a feeling that it will, as long as the reading happens on the same millisecond. Wants checking though.

File: mijia-homie/src/mijia-history-influx.rs (hunk 3 above)
Comment (qwandor) [reply mijia-homie/src/mijia-history-influx.rs:91]: Yes.

File: mijia-homie/src/mijia-history-influx.rs (hunk 1 above)
Comment (qwandor) [reply mijia-homie/src/mijia-history-influx.rs]: Done.

File: mijia-homie/src/mijia-history-influx.rs (hunk 2 above)
Comment (qwandor) [reply mijia-homie/src/mijia-history-influx.rs]: Added a check with a configurable maximum.

**Comment by qwandor** [comment]
> If you run it as a cronjob every hour, will influx delicate the readings correctly? I have a feeling that it will, as long as the reading happens on the same millisecond. Wants checking though.

It appears that InfluxDB does indeed deduplicate the data.

**Pull request merged**